src/goto-symex/builtin_functions.cpp

```diff
@@ -1027,3 +1027,24 @@ void goto_symext::intrinsic_memset(
     bump_call();
   }
 }
+
+void goto_symext::intrinsic_get_object_size(
+  const code_function_call2t &func_call,
+  reachability_treet &)
+{
+  assert(func_call.operands.size() == 1 && "Wrong get_object_size signature");
+  auto ptr = func_call.operands[0];
+
+  // Work out what the ptr points at.
+  internal_deref_items.clear();
+  dereference2tc deref(get_empty_type(), ptr);
+  dereference(deref, dereferencet::INTERNAL);
+
+  assert(is_array_type(internal_deref_items.front().object->type));
+  auto obj_size =
+    to_array_type(internal_deref_items.front().object->type).array_size;
```

Comment:
Is it only ever one/the first object?

Reply:
It should be, the dereference should only return the memory symbol.

Reply:
So the assumption are a) that `__ESBMC_get_object_size` is only applied to pointers returned from `malloc`, `new`, etc. and never to pointers to anything else and b) that the value-set of that pointer is a singleton? If so, could we maybe document these assumption (since they are not clear from the signature of the intrinsic)?